workbook/serializers/worker_serializer.py

```diff
@@ -0,0 +1,118 @@
+from django.core.exceptions import BadRequest
+from rest_framework import serializers
+
+from workbook.models.models import *
+from workbook.serializers.common_serializer import UserSerializer, ReviewSerializer
+
+
+class WorkerSerializer(serializers.Serializer):
+    profile_picture = serializers.ImageField()
+    biography = serializers.CharField(max_length=300)
+    job_title = serializers.CharField(max_length=50)
+    day_start_time = serializers.TimeField()
+    day_end_time = serializers.TimeField()
+
+    def validate_profile_picture(self, value):
+        if not value.name.endswith(('jpg', 'jpeg', 'png')):
+            raise serializers.ValidationError("Only JPG, JPEG, and PNG files are allowed.")
+        if value.size > 2 * 1024 * 1024:  # 2MB limit
+            raise serializers.ValidationError("Image size should be under 2MB.")
+        return value
+
+    def validate_biography(self, value):
+        if len(value.strip()) == 0:
+            raise serializers.ValidationError("Biography cannot be empty.")
+        if len(value) > 300:
+            raise serializers.ValidationError("Biography cannot exceed 300 characters.")
+        return value
+
+    def validate_job_title(self, value):
+        if len(value.strip()) == 0:
+            raise serializers.ValidationError("Job title cannot be empty.")
+        if len(value) > 50:
+            raise serializers.ValidationError("Job title cannot exceed 50 characters.")
+        return value
+
+    def validate(self, data):
+        day_start_time = data.get('day_start_time')
+        day_end_time = data.get('day_end_time')
+
+        if day_start_time and day_end_time and day_end_time <= day_start_time:
+            raise serializers.ValidationError("End time must be after start time.")
+
+        return data
+
+    def is_valid_raise(self, raise_exception=True):
+        if not self.is_valid():
+            if raise_exception:
+                raise BadRequest(self.errors)
+            return False
+        return True
+
+
+class WorkerReservationSerializer(serializers.ModelSerializer):
+    # source='review_set' means Django pulls all the Review objects that are linked to each Reservation.
+    reviews = ReviewSerializer(many=True, source='review_set')
```

Comment:
if it will cause lazy loading then do it in the service layer and eagerly fetch all the objects